src/titiler/extensions/titiler/extensions/render.py

```diff
@@ -0,0 +1,182 @@
+                factory.dataset_dependency,
+                # Image rendering Dependencies
+                factory.rescale_dependency,
+                factory.color_formula_dependency,
+                factory.colormap_dependency,
+                factory.render_dependency,
+            ]
+
+            query = {"url": src_path}
+            for dependency in tile_dependencies:
+                dep = get_dependant(path="", call=dependency)
+                if dep.query_params:
+                    # call the dependency with the query-parameters values
+                    query_values, _errors = request_params_to_args(
+                        dep.query_params, render
+                    )
+                    _ = dependency(**query_values)
+                    query.update(query_values)
+            return urlencode(
+                {key: value for key, value in query.items() if value is not None},
+                doseq=True,
+            )
+
+        def _prepare_render_item(
+            render_id: str, render: Dict, request: Request, src_path: str
+        ) -> Dict:
+            """Prepare single render item."""
+            query_string = _prepare_query_string(render, src_path)
+            links = [
+                {
+                    "href": factory.url_for(
+                        request,
+                        "Show STAC render",
+                        render_id=render_id,
+                    )
+                    + "?"
+                    + urlencode({"url": src_path}),
+                    "rel": "self",
+                    "type": "application/json",
+                    "title": f"{render_id} render item",
+                },
+                {
+                    "href": factory.url_for(
+                        request,
+                        "tilejson",
+                        tileMatrixSetId="{tileMatrixSetId}",
+                    )
+                    + "?"
+                    + query_string,
+                    "rel": "tilesets-map",
+                    "title": f"tilejson file for {render_id}",
+                    "templated": True,
+                },
+                {
+                    "href": factory.url_for(
+                        request,
+                        "wmts",
+                        tileMatrixSetId="{tileMatrixSetId}",
+                    )
+                    + "?"
+                    + query_string,
+                    "rel": "tilesets-map",
+                    "title": f"WMTS service for {render_id}",
+                    "templated": True,
+                },
+            ]
+
+            return {
+                "params": render,
+                "links": links,
+            }
+
+        @factory.router.get(
+            "/renders",
+            response_model=RenderItemList,
+            response_model_exclude_none=True,
+            name="List STAC renders",
+        )
+        def render_list(request: Request, src_path=Depends(factory.path_dependency)):
```

Comment:
let's add `reader_params=Depends(factory.reader_dependency),`

https://github.com/developmentseed/titiler/blob/22e48f0c17784de5c32900b789cc4acd1aab9fb3/src/titiler/core/titiler/core/factory.py#L1414-L1420

I think it's fine not having the `env` dependency because we won't access the raster